keyboards/junco/junco_sync.c

```diff
@@ -0,0 +1,286 @@
+// Copyright 2022 Dane Skalski (@Daneski13)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+/* Only relevant if junco sync is enabled */
+#ifdef JUNCO_SYNC_ENABLE
+#    include "junco_sync.h"
+#    include "print.h"
+#    include "transactions.h"
+#    include "split_util.h"
+#    include "junco.h"
+
+/* Sync common keyboard data */
+
+// Pointer to the user's user_config.raw. If the user wants to sync their own user config,
+// set this to the address of user_config.raw in the keyboard_post_init after reading the
+// user_config from eeprom `user_config_raw_ptr = &user_config.raw`.
+uint32_t* user_config_raw_ptr = NULL;
+
+// Data type for sending data to slave
+typedef struct {
+    uint32_t kb_conf;     // Current keyboard config
+    uint32_t default_lay; // Current default layer, SPLIT_LAYER_STATE_ENABLE is not persistent and may not sync in time
```

Comment:
I'm curious about this. I've not seen that be an issue.  But if you're writing to eeprom when doing this, it can slow down things.

Reply:
In testing the layer state rarely synced in time for the slave to write the new default layer to EEPROM when the slave handler was called, thus I'm just explicitly sending the default layer to the slave and writing to EEPROM that way.